src/pages/home/report/ReportActionsView.tsx

```diff
@@ -164,81 +221,149 @@ function ReportActionsView({
+
+            // Capture the init measurement only once not per each chat switch as the value gets overwritten
+            if (!ReportActionsView.initMeasured) {
+                Performance.markEnd(CONST.TIMING.REPORT_INITIAL_RENDER);
+                ReportActionsView.initMeasured = true;
+            } else {
+                Performance.markEnd(CONST.TIMING.SWITCH_REPORT);
+            }
+        },
+        [hasCachedActions],
+    );
+
+    useEffect(() => {
+        // Temporary solution for handling REPORTPREVIEW. More details: https://expensify.slack.com/archives/C035J5C9FAP/p1705417778466539?thread_ts=1705035404.136629&cid=C035J5C9FAP
+        // This code should be removed once REPORTPREVIEW is no longer repositioned.
+        // We need to call openReport for gaps created by moving REPORTPREVIEW, which causes mismatches in previousReportActionID and reportActionID of adjacent reportActions. The server returns the correct sequence, allowing us to overwrite incorrect data with the correct one.
+        const shouldOpenReport =
+            firstReportActionName === CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW &&
+            !hasCreatedAction &&
+            isReadyForCommentLinking &&
+            reportActions.length < 24 &&
+            reportActions.length >= 1 &&
+            !isLoadingInitialReportActions &&
+            !isLoadingOlderReportActions &&
+            !isLoadingNewerReportActions;
+
+        if (shouldOpenReport) {
+            Report.openReport(reportID, reportActionID);
         }
+    }, [
+        hasCreatedAction,
+        reportID,
+        reportActions,
+        reportActionID,
+        firstReportActionName,
+        isReadyForCommentLinking,
+        isLoadingOlderReportActions,
+        isLoadingNewerReportActions,
+        isLoadingInitialReportActions,
+    ]);
+
+    // Check if the first report action in the list is the one we're currently linked to
+    const isTheFirstReportActionIsLinked = firstReportActionID === reportActionID;
 
-        didLayout.current = true;
-        Timing.end(CONST.TIMING.SWITCH_REPORT, hasCachedActions ? CONST.TIMING.WARM : CONST.TIMING.COLD);
+    useEffect(() => {
+        let timerId: NodeJS.Timeout;
 
-        // Capture the init measurement only once not per each chat switch as the value gets overwritten
-        if (!ReportActionsView.initMeasured) {
-            Performance.markEnd(CONST.TIMING.REPORT_INITIAL_RENDER);
-            ReportActionsView.initMeasured = true;
+        if (isTheFirstReportActionIsLinked) {
+            setIsInitialLinkedView(true);
         } else {
-            Performance.markEnd(CONST.TIMING.SWITCH_REPORT);
+            // After navigating to the linked reportAction, apply this to correctly set
+            // `autoscrollToTopThreshold` prop when linking to a specific reportAction.
+            InteractionManager.runAfterInteractions(() => {
+                // Using a short delay to ensure the view is updated after interactions
```

Comment:
Do we need both `InteractionManager.runAfterInteractions` _and_ the timeout, or can we remove the timeout?

Reply:
This fix is pretty tricky. From my observations, we just need a tad more time before we can turn on `autoscrollToTopThreshold` without any hitches. I was thinking of stretching out the timeout to 100-200 ms, but that feels like overkill. Using nested InteractionManager.runAfterInteractions won't work either. I'll look for a more elegant solution to address this.